featuremanagement/_featuremanager.py

```diff
@@ -64,6 +64,8 @@ class FeatureManager:
     :type configuration: Mapping
     :keyword feature_filters: Custom filters to be used for evaluating feature flags
     :paramtype feature_filters: list[FeatureFilter]
+    :keyword on_feature_evaluated: Callback function to be called when a feature flag is evaluated
```

Comment:
I think we may want to rethink this callback function. This makes sense with the publisher idea- but OpenTelemetry's whole goal is to stop custom callbacks like this. It might make sense to use their api to push an event here instead and let an exporter handle pushing the event to app insights/whereever. 

In .net- I'm working on a design doc now to adjust to this approach.